contracts/extensions/contracts/MilestonesExtension.sol

```diff
@@ -0,0 +1,218 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity 0.8.19;
+
+// Interfaces
+import {IMilestonesExtension} from "../interfaces/IMilestonesExtension.sol";
+import {IRegistry} from "../../core/interfaces/IRegistry.sol";
+// Core Contracts
+import {CoreBaseStrategy} from "../../strategies/CoreBaseStrategy.sol";
+// Internal Libraries
+import {Metadata} from "../../core/libraries/Metadata.sol";
+
+/// @title Milestone Extension Strategy
+abstract contract MilestonesExtension is CoreBaseStrategy, IMilestonesExtension {
+    /// ================================
+    /// ========== Storage =============
+    /// ================================
+
+    /// @notice The maximum bid allowed.
+    uint256 public maxBid;
+
+    /// @notice The upcoming milestone which is to be paid.
+    uint256 public upcomingMilestone;
+
+    /// @notice This maps recipients to their bids
+    /// @dev 'recipientId' to 'bid'
+    mapping(address => uint256) public bids;
+
+    /// @notice Collection of milestones
+    Milestone[] internal milestones;
+
+    /// ===============================
+    /// ========= Initialize ==========
+    /// ===============================
+
+    /// @notice This initializes the BaseStrategy
+    /// @dev You only need to pass the 'poolId' to initialize the BaseStrategy and the rest is specific to the strategy
+    /// @param _initializeParams The initialize params
+    function __MilestonesExtension_init(InitializeParams memory _initializeParams) internal {
+        // Set the strategy specific variables
+        _increaseMaxBid(_initializeParams.maxBid);
```

Comment:
since we only make use of maxBid, should we just pass as param that instead of the whole InitializeParams type?